src/app.service.ts

```diff
@@ -1,8 +1,21 @@
 import { Injectable } from '@nestjs/common';
+import { DataSource } from 'typeorm';
 
 @Injectable()
 export class AppService {
+  constructor(private readonly dataSource: DataSource) {}
   getHello(): string {
     return 'Hello World!';
   }
+
+  async getTotalSalesForYear(year: number): Promise<number> {
+    const query = `
+      SELECT SUM(TotalAmount)
+      FROM Orders
+      WHERE EXTRACT(YEAR FROM OrderDate) = $1
+    `;
+
+    const result = await this.dataSource.query(query, [year]);
```

Comment:
Consider adding error handling for the `getTotalSalesForYear` method to catch and manage potential database query errors. This can help prevent unexpected application crashes and provide meaningful error messages. For example, wrap the query execution in a try-catch block and log or handle the error appropriately. [important]